pdns/dnsdistdist/dnsdist-configuration.hh

```diff
@@ -0,0 +1,179 @@
+/*
+ * This file is part of PowerDNS or dnsdist.
+ * Copyright -- PowerDNS.COM B.V. and its contributors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * In addition, for the avoidance of any doubt, permission is granted to
+ * link this program with OpenSSL and to (re)distribute the binaries
+ * produced as the result of such linking.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#pragma once
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "config.h"
+#include "credentials.hh"
+#include "dnsdist-actions.hh"
+#include "dnsdist-carbon.hh"
+#include "dnsdist-query-count.hh"
+#include "dnsdist-rule-chains.hh"
+#include "iputils.hh"
+
+class ServerPolicy;
+struct ServerPool;
+struct DownstreamState;
+struct ClientState;
+
+using servers_t = std::vector<std::shared_ptr<DownstreamState>>;
+
+namespace dnsdist::configuration
+{
+/* when we add EDNS to a query, we don't want to advertise
+   a large buffer size */
+static constexpr size_t s_EdnsUDPPayloadSize{512};
+static constexpr uint16_t s_defaultPayloadSizeSelfGenAnswers = 1232;
+static constexpr uint16_t s_udpIncomingBufferSize{1500}; // don't accept UDP queries larger than this value
+static_assert(s_defaultPayloadSizeSelfGenAnswers < s_udpIncomingBufferSize, "The UDP responder's payload size should be smaller or equal to our incoming buffer size");
+
+/* this part of the configuration can only be updated at configuration
+   time, and is immutable once the configuration phase is over */
+struct Configuration
```

Comment:
I would prefer this (it will cause a lot of churn though...)
```suggestion
struct ImmutableConfiguration
```

Reply:
Good one! I just did it, it was not as bad as I expected, and better now than later :)